wagtail/admin/views/generic/preview.py

```diff
@@ -163,3 +166,55 @@ def get(self, request, *args, **kwargs):
             raise PermissionDenied
 
         return self.revision_object.make_preview_request(request, preview_mode)
+
+
+@method_decorator(xframe_options_sameorigin_override, name="get")
+class StreamFieldBlockPreview(TemplateView):
+    template_name = "wagtailadmin/generic/streamfield_block_preview.html"
+    http_method_names = ("get",)
+
+    @cached_property
+    def block_id(self):
+        return self.request.GET.get("id")
+
+    @cached_property
+    def block_def(self) -> Block:
+        if not (block := Block.definition_registry.get(self.block_id)):
+            raise Http404
+        return block
+
+    @cached_property
+    def block_value(self):
+        return self.block_def.get_preview_value()
+
+    @cached_property
+    def page_title(self):
+        return gettext("Preview for %(block_label)s (%(block_type)s)") % {
+            "block_label": self.block_def.label,
+            "block_type": self.block_def.__class__.__name__,
+        }
+
+    @cached_property
+    def base_context(self):
+        # Do NOT use the name `block` in the context, as it will conflict with
+        # the current block inside a template {% block %} tag.
+        return {
+            "request": self.request,
+            "block_def": self.block_def,
+            "block_class": self.block_def.__class__,
+            "bound_block": self.block_def.bind(self.block_value),
+            "page_title": self.page_title,
+        }
+
+    def get_template_names(self):
+        templates = [self.template_name]
+        if preview_template := self.block_def.get_preview_template(
+            self.block_value, self.base_context
+        ):
+            templates.insert(0, preview_template)
+        return templates
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context.update(self.base_context)
+        return self.block_def.get_preview_context(self.block_value, context)
```

Comment:
I'm not sure whether `get_preview_context` should provide the final context, or whether it should only `.update()` the existing view context.